src/commonMain/kotlin/fr/acinq/lightning/serialization/v4/Deserialization.kt

```diff
@@ -99,6 +100,16 @@ object Deserialization {
                 origins = readCollection { readChannelOrigin() as Origin.PayToOpenOrigin }.toList()
             )
             else -> error("unknown discriminator $discriminator for class ${SpliceStatus::class}")
+        },
+        liquidityPurchases = when {
+            availableBytes == 0 -> listOf()
+            else -> when (val discriminator = read()) {
+                0x01 -> {
+                    val liquidityPurchaseCount = readNumber()
+                    (0 until liquidityPurchaseCount).map { readLiquidityPurchase() }
+                }
```

Comment:
```suggestion
                0x01 -> readCollection { readLiquidityPurchase() }.toList()
```

Reply:
Done in https://github.com/ACINQ/lightning-kmp/pull/561/commits/f735f45eee0b765ed04563ecfad0f5a0b061a774